apps/dcos_l4lb/src/dcos_l4lb_mesos_poller.erl

```diff
@@ -144,11 +161,17 @@ collect_vips(Tasks) ->
 -spec(collect_vips(task_id(), task(), VIPs) -> VIPs
     when VIPs :: #{key() => [backend()]}).
 collect_vips(TaskId, Task, VIPs) ->
-    lists:foldl(fun (Port, Acc) ->
+    VIPs1 = lists:foldl(fun (Port, Acc) ->
         lists:foldl(fun (VIPLabel, Bcc) ->
             collect_vips(TaskId, Task, Port, VIPLabel, Bcc)
         end, Acc, maps:get(vip, Port, []))
-    end, VIPs, maps:get(ports, Task, [])).
+    end, VIPs, maps:get(ports, Task, [])),
+
+    BackendsCount = lists:foldl(fun (BEs, TotalBEs) ->
+        TotalBEs + length(BEs)
+    end, 0, maps:values(VIPs1)),
+    prometheus_gauge:set(l4lb, local_backends, [], BackendsCount),
```

Comment:
My suggestion is to move `local_backends` metric from  `collect_vips/3` function to `handle_poll_state/1`. Also it can be simplified:

```
    prometheus_gauge:set(
        l4lb, local_backends, [],
        lists:sum([length(BEs) || BEs <- maps:values(VIPs)]))
```